radicle-node/src/client.rs

```diff
@@ -103,10 +104,14 @@ impl<G: crypto::Signer + crypto::Negotiator + Clone + 'static> Runtime<G> {
             rng,
         );
 
+        let node_id = signer.public_key().0;
+        let sig = signer.sign(node_id.as_slice());
+        let auth = Authenticator::new(node_id.into(), sig.0.into());
+
```

Comment:
We'll need a nonce too actually!

Reply:
To save a round-trip, maybe the nonce could be the noise session hash :thinking: